### PR TITLE
[PA-213] [PA-225] [PA-227] New system tests

### PR DESCRIPTION
**Pull request by bhsrinivasan-px**
**What this PR does / why we need it**:
Implements all of the new system tests from tickets PA-213, PA-225, PA-227

**Which issue(s) this PR fixes** (optional)
Closes # PA-213, PA-225, PA-227

**Special notes for your reviewer**:
CI run: https://jenkins.pwx.dev.purestorage.com/job/PDS/job/pds-branch-build-test/48/console

### CODE REVIEW

File: tests/pds/dataservice_test.go [hunk 1]
```diff
@@ -614,3 +617,437 @@ func UpgradeDataService(dataservice, oldVersion, oldImage, dsVersion, dsBuild st
...
+			cleanup = append(cleanup, deployment.GetId())
+
+		})
+
+	}
+
+	return cleanup
+}
+
+var _ = Describe("{DeployMultipleNamespaces}", func() {
```
Comment (Madanagopal19) [inline tests/pds/dataservice_test.go:687]: Add the below statements in start of the test cases 
`JustBeforeEach(func() {
		StartTorpedoTest("DeletePDSPods", "delete pds pods and validate if its coming back online and dataserices are not affected", nil, 0)
	})`

Add the below lines at end of the tests
`JustAfterEach(func() {
		defer EndTorpedoTest()
	})`

File: tests/pds/dataservice_test.go [hunk 2]
```diff
@@ -614,3 +617,437 @@ func UpgradeDataService(dataservice, oldVersion, oldImage, dsVersion, dsBuild st
...
+
+	It("Deploy Dataservices and delete namespace", func() {
+
+		nname := "test-namespace-0"
+		_, err := pdslib.CreateK8sPDSNamespace(nname)
+		logrus.Infof("Created namespace: %v", nname)
+		dash.Infof("Created namespace: %v", nname)
+		Expect(err).NotTo(HaveOccurred())
+
+		defer func() {
```
Comment (Madanagopal19) [inline tests/pds/dataservice_test.go:752]: delete this block if not required

File: tests/pds/dataservice_test.go [hunk 3]
```diff
@@ -614,3 +617,437 @@ func UpgradeDataService(dataservice, oldVersion, oldImage, dsVersion, dsBuild st
...
+	})
+
+})
+
+var _ = Describe("{DeletePDSEnabledNamespace}", func() {
+
+	It("Deploy Dataservices and delete namespace", func() {
+
+		nname := "test-namespace-0"
+		_, err := pdslib.CreateK8sPDSNamespace(nname)
```
Comment (Madanagopal19) [inline tests/pds/dataservice_test.go:747]: handle the error, then use log statements

File: tests/pds/dataservice_test.go [hunk 4]
```diff
@@ -614,3 +617,437 @@ func UpgradeDataService(dataservice, oldVersion, oldImage, dsVersion, dsBuild st
...
+				}
+				isDeploymentsDeleted = true
+			})
+
+		})
+
+	})
+})
+
+var _ = Describe("{RestartPXPods}", func() {
```
Comment (Madanagopal19) [inline tests/pds/dataservice_test.go:869]: same as above, include before and after each methods

File: tests/pds/dataservice_test.go [hunk 5]
```diff
@@ -614,3 +617,437 @@ func UpgradeDataService(dataservice, oldVersion, oldImage, dsVersion, dsBuild st
...
+						nodeObject, err := pdslib.GetK8sNodeObjectUsingPodName(pod.Spec.NodeName)
+						Expect(err).NotTo(HaveOccurred())
+						nodeList = append(nodeList, nodeObject)
+					}
+				})
+
+				Step("For each node in the nodelist, stop px service on it", func() {
+
+					for _, node := range nodeList {
+						label := "px/service=stop"
```
Comment (Madanagopal19) [inline tests/pds/dataservice_test.go:990]: "px/service=stop" make it as a constant

File: tests/pds/dataservice_test.go [hunk 6]
```diff
@@ -614,3 +617,437 @@ func UpgradeDataService(dataservice, oldVersion, oldImage, dsVersion, dsBuild st
...
+				})
+
+				Step("Validate that the deployment is healthy", func() {
+					err := pdslib.ValidateDataServiceDeployment(deployment, namespace)
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				Step("Cleanup: Start px on the node and uncordon the node", func() {
+					for _, node := range nodeList {
+						label := "px/service"
```
Comment (Madanagopal19) [inline tests/pds/dataservice_test.go:1008]: make it as a constant

**Review by Madanagopal19 — CHANGES_REQUESTED** [review]

File: drivers/pds/lib/utils.go [hunk 7]
```diff
@@ -1358,3 +1359,165 @@ func ValidateAllDataServiceVolumes(deployment *pds.ModelsDeployment, dataService
...
+	if err != nil {
+		logrus.Errorf("The namespace %v has not been deleted", namespace)
+		return fmt.Errorf("the namespace %v has not been deleted", namespace)
+	}
+	logrus.Infof("The namespace has been successfully deleted")
+	return nil
+
+}
+
+func CreateK8sPDSNamespace(nname string) (*corev1.Namespace, error) {
```
[the remark this answers is not in the document]
Comment (bhsrinivasan-px) [reply drivers/pds/lib/utils.go:1409]: would require a lot of refactoring, will add a TODO

File: tests/pds/dataservice_test.go [hunk 8]
```diff
@@ -614,3 +617,437 @@ func UpgradeDataService(dataservice, oldVersion, oldImage, dsVersion, dsBuild st
 		}()
 	})
 }
+
+func DeployInANamespaceAndVerify(nname string, namespaceID string) []string {
```
[the remark this answers is not in the document]
Comment (bhsrinivasan-px) [reply tests/pds/dataservice_test.go:621]: done

File: tests/pds/dataservice_test.go (hunk 1 above)
Comment (bhsrinivasan-px) [reply tests/pds/dataservice_test.go:687]: done

File: tests/pds/dataservice_test.go (hunk 2 above)
Comment (bhsrinivasan-px) [reply tests/pds/dataservice_test.go:752]: done

File: tests/pds/dataservice_test.go (hunk 4 above)
Comment (bhsrinivasan-px) [reply tests/pds/dataservice_test.go:869]: done

File: tests/pds/dataservice_test.go (hunk 3 above)
Comment (bhsrinivasan-px) [reply tests/pds/dataservice_test.go:747]: done

**Review by Madanagopal19 — APPROVED** [review]
LGTM

**Pull request merged**